src/FrameSemantics.cc

```diff
@@ -282,289 +282,511 @@ static Errors resolveModelPoseWithPlacementFrame(
   return errors;
 }
 
-/////////////////////////////////////////////////
-Errors buildFrameAttachedToGraph(
-    ScopedGraph<FrameAttachedToGraph> &_out, const Model *_model, bool _isRoot)
+/// \brief Base struct the contains a few common members. These structs provide
+/// a common API used by the build*Graph functions to access the various
+/// attributes and retrieve children of regular DOM objects and Interface
+/// Elements.
+struct WrapperBase
 {
-  Errors errors;
-
-  if (!_model)
-  {
-    errors.push_back({ErrorCode::ELEMENT_INVALID,
-        "Invalid sdf::Model pointer."});
-    return errors;
-  }
-  else if (!_model->Element())
+  /// \brief Name of the entity
+  const std::string name;
+  /// \brief Element type, such as Model, Link, or Interface Frame.
+  const std::string elementType;
+  /// \brief Frame type used int the FrameAttachedTo or PoseRelativeTo graphs.
```

Comment:
used *in* the

Reply:
Done in https://github.com/ignitionrobotics/sdformat/pull/764/commits/59050f96134d6bf0cea07e35ab7916b1da7b70d9.